### PR TITLE
v1.5 final release: quick hits

### LINKED ISSUE #3200: Model contracts: examples with size or precision & scale
### Contributions

- [X] I have read the contribution docs, and understand what's expected of me.

### Link to the page on docs.getdbt.com requiring updates

https://docs.getdbt.com/docs/collaborate/govern/model-contracts

### What part(s) of the page would you like to see updated?

Within the database, certain data types can have size specified (e.g., `varchar(80)`). Others can have precision and scale specified (e.g., `decimal(18, 2)`). Others will only have precision (e.g. `timestamp_tz(3)`). Others will have nothing extra (e.g., `bool`, `int`, etc).

We should make it clear through examples how model contracts will treat these types of columns.

e.g. What will happen if the model contract says `varchar(80)` but the actual database type is `varchar(256)`?

My belief is that the model contact will allow this to pass the pre-flight check and that this is the intended behavior. This could surprise some folks, so we'll want to be explicit in our docs.

### Additional information

Related:
https://docs.getdbt.com/reference/dbt-classes#static-methods

### LINKED ISSUE #3201: Custom implementations of `generate_alias_name macro`
### Contributions

- [X] I have read the contribution docs, and understand what's expected of me.

### Link to the page on docs.getdbt.com requiring updates

https://docs.getdbt.com/guides/migration/versions/upgrading-to-v1.5#behavior-changes

### What part(s) of the page would you like to see updated?

The bottom of this page mentions some crucial information regarding the `generate_alias_name ` macro that should be included in the upgrade guide:
https://docs.getdbt.com/docs/collaborate/govern/model-versions#how-to-create-a-new-version-of-a-model

Namely:
> Projects which have historically implemented [custom aliases](https://docs.getdbt.com/docs/build/custom-aliases) by reimplementing the `generate_alias_name` macro will need to update their custom implementations to account for model versions.

This should go in the "behavior changes" section.

### Additional information

_No response_

### LINKED ISSUE #3207: `--populate-cache`/`--no-populate-cache` config
### Contributions

- [X] I have read the contribution docs, and understand what's expected of me.

### Link to the page on docs.getdbt.com requiring updates

https://docs.getdbt.com/reference/global-configs

### What part(s) of the page would you like to see updated?

`--populate-cache/--no-populate-cache`

Totally skip upfront population of relational cache. This is most useful when running only a few models, or interactively compiling/previewing a query. Similar use case to `--cache-selected-only`.

dbt will still run caching queries as needed, and populate the cache with the results of those queries.

### Additional information

- https://github.com/dbt-labs/dbt-core/issues/6526
- https://github.com/dbt-labs/dbt-core/pull/7307

### LINKED ISSUE #3212: Clarify that `dbt show` always reruns compiled code, and does not sample already-materialized model
### Contributions

- [X] I have read the contribution docs, and understand what's expected of me.

### Link to the page on docs.getdbt.com requiring updates

https://docs.getdbt.com/reference/commands/show

### What part(s) of the page would you like to see updated?

> `dbt show` generates post-transformation preview from a source model, test, or analysis file.

The words "post-transformation" here are misleading IMO. This implies that the model has already been transformed (= materialized), and we're just previewing the already-existing table. While there are compelling reasons to want this, such as the ability to take advantage of [BQ's zero-cost "preview data" API](https://cloud.google.com/bigquery/docs/best-practices-costs#preview-data) (`bq head`) — and maybe it should be a configuration option in the future, or another command (`dbt head`?) — it's not the current intended behavior of `dbt show`.

Let's make clear that `dbt show --select specific_model` does exactly the following:
- Compiles the dbt-SQL for `specific_model`
- Executes that SQL
- Shows a preview of it on the CLI

That's true regardless of whether `specific_model` has just been run, has been run recently, has never been run, now differs in its logical (SQL definition) versus applied (db object) state, ...

### Additional information

_No response_

### LINKED ISSUE #3236: Add to v1.5 migration guide: changed data type of `refs` in manifest
### Contributions

- [X] I have read the contribution docs, and understand what's expected of me.

### Link to the page on docs.getdbt.com requiring updates

https://docs.getdbt.com/guides/migration/versions/upgrading-to-v1.5

### What part(s) of the page would you like to see updated?

Changed data type of `refs`:
- previous type: array of arrays of string (`List[List[str]]`)
- new type: array of `RefArgs(name: str, package: Optional[str] = None, version: Union[str, float, NoneType] = None)`

### Additional information

_No response_

### LINKED ISSUE #3241: Clarify distinctions between model-level groups/access and dbt Cloud RBAC
### Contributions

- [X] I have read the contribution docs, and understand what's expected of me.

### Link to the page on docs.getdbt.com requiring updates

Model-level access in `dbt-core`:
- https://docs.getdbt.com/docs/collaborate/govern/model-access
- https://docs.getdbt.com/reference/resource-properties/access

Role-based access in dbt Cloud:
https://docs.getdbt.com/docs/cloud/manage-access/about-access

### What part(s) of the page would you like to see updated?

Let's disambiguate between these two concepts, by very clearly calling out that they are not the same thing:
- Model-level access, for managing cross-references within projects
- Access for people using/managing resources in dbt Cloud

Same goes for:
- Groups of models in a dbt project
- Groups of people/users in dbt Cloud

Our intent is for these concepts to be related & complementary — especially later this year as we build differentiated experiences into dbt Cloud for multi-project deployments — but they are definitely not the same thing!

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by jtcohen6**
resolves #3241
resolves #3201
resolves #3236
resolves #3207
resolves #3200
resolves #3212

## Checklist
Uncomment if you're publishing docs for a prerelease version of dbt (delete if not applicable):
- [x] Add versioning components, as described in [Versioning Docs](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#versioning-entire-pages)
- [x] Add a note to the prerelease version [Migration Guide](https://github.com/dbt-labs/docs.getdbt.com/tree/current/website/docs/guides/migration/versions)

Removing or renaming existing pages (delete if not applicable):
- [x] Remove page from `website/sidebars.js`
- [x] Add an entry `_redirects`
- [x] [Ran link testing](https://github.com/dbt-labs/docs.getdbt.com#running-the-cypress-tests-locally) to update the links that point to the deleted page

### CODE REVIEW

File: website/docs/guides/migration/versions/02-upgrading-to-v1.5.md
Comment (dbeatty10) [inline website/docs/guides/migration/versions/02-upgrading-to-v1.5.md]: Indeed, it's not a _strict_ requirement before upgrading.

But it feels like a ticking time bomb if they don't update their customized `generate_alias_name` as part of their upgrade. 💣 

i.e., if they upgrade now but don't try to start using model versions until 6+ months from now, they could really be scratching their heads trying to figure out why things are not working as documented. 💥

File: website/docs/reference/global-configs.md
Comment (dbeatty10) [inline website/docs/reference/global-configs.md]: I don't know what our style guide says about terminal commands beginning with `$` or not.

But I prefer _without_ because it makes it easier for me to copy-n-paste the entire block without having to carefully exclude the leading character.

In fact, I'd vote for doing reviewing all our code examples throughout docs.getdbt.com and removing all the `$`! 🤑

File: website/docs/docs/cloud/manage-access/about-access.md
Comment (mirnawong1) [inline website/docs/docs/cloud/manage-access/about-access.md]:
```suggestion
:::info User access is not Model access
```

File: website/docs/reference/resource-configs/contract.md
Comment (dbeatty10) [inline website/docs/reference/resource-configs/contract.md]: > inside and outside dbt

This is money 🎯

File: website/docs/reference/resource-configs/contract.md
Comment (dbeatty10) [inline website/docs/reference/resource-configs/contract.md]: > dbt will **not** include size, precision, or scale in its data type comparison

This is a key insight! 👀

File: website/docs/docs/cloud/manage-access/about-access.md
Comment (mirnawong1) [inline website/docs/docs/cloud/manage-access/about-access.md]:
```suggestion
**User groups and access** and **model groups and access** mean two different things. "Model groups and access" is a specific term used in the language of dbt-core. Refer to [Model access](/docs/collaborate/govern/model-access)  for more info on what it means in dbt-core.
```

File: website/docs/docs/cloud/manage-access/about-access.md
Comment (mirnawong1) [inline website/docs/docs/cloud/manage-access/about-access.md]:
```suggestion
```

**Review by dbeatty10 — DISMISSED** [review]
I'd say ship it! :shipit:

Left a couple comments.

Caveats:
- Although I read every single word of this PR, I did **not** verify any links
- I only read the text, I did **not** examine any Netlify previews of the actual rendering

File: website/docs/docs/collaborate/govern/model-access.md
Comment (mirnawong1) [inline website/docs/docs/collaborate/govern/model-access.md]:
```suggestion
:::info Model access is not User access
```

File: website/docs/docs/collaborate/govern/model-access.md
Comment (mirnawong1) [inline website/docs/docs/collaborate/govern/model-access.md:14]: **Model groups and access** and **user groups and access** mean two different things. "User groups and access" is a specific term used in dbt Cloud. Refer to [User access](/docs/cloud/manage-access/about-user-accessl)  for more info.

Comment (mirnawong1) [inline website/docs/docs/collaborate/govern/model-access.md]:
```suggestion
**Model groups and access** and **user groups and access** mean two different things. "User groups and access" is a specific term used in dbt Cloud. Refer to [User access](https://github.com/docs/cloud/manage-access/about-user-accessl) for more info.
```

File: website/docs/docs/collaborate/govern/model-access.md
Comment (mirnawong1) [inline website/docs/docs/collaborate/govern/model-access.md]:
```suggestion
- It enables you to designate certain models as "private" for use exclusively within a particular group.
```

File: website/docs/docs/collaborate/govern/model-access.md
Comment (mirnawong1) [inline website/docs/docs/collaborate/govern/model-access.md]:
```suggestion
Some models are implementation details, meant for reference only within their group of related models. Other models should be accessible through the [ref](/reference/dbt-jinja-functions/ref) function across groups and projects. Models can set an [access modifier](https://en.wikipedia.org/wiki/Access_modifiers) to indicate their intended level of accessibility.
```

**Comment by jtcohen6** [comment]
Note to self:
- also clarify that model access != database permissions (at least not implicitly / auto-magically coordinated)
- show that `groups` can be configured in `dbt_project.yml` for an entire subfolder of models at once

**Comment by dbeatty10** [comment]
> clarify that model access != database permissions (at least not implicitly / auto-magically coordinated)

Gonna need to beat this drum early, often, and everywhere folks use `grants` and/or `groups`.

File: website/docs/docs/collaborate/govern/model-contracts.md
Comment (mirnawong1) [inline website/docs/docs/collaborate/govern/model-contracts.md]: i'd recommend being more clear on which platforms are supported here. what are 'popular ones'? also should we link to the supported data platform page?

```suggestion
- Most commonly used [data platforms](/docs/supported-data-platforms), such as Snowflake, BigQuery, Databricks, Redshift, and so on, support `constraints`. However, the specific `constraints` that are supported or enforced can vary depending on the platform.
```

File: website/docs/docs/collaborate/govern/model-contracts.md
Comment (mirnawong1) [inline website/docs/docs/collaborate/govern/model-contracts.md]: non blocker but this section almost feels like it should be on the top as part of the intro. not really a prerequisites

File: website/docs/docs/collaborate/govern/model-contracts.md
Comment (mirnawong1) [inline website/docs/docs/collaborate/govern/model-contracts.md]:
```suggestion
It's important to define contracts for "public" models that are shared with other groups, teams, and (soon) dbt projects. Refer to [Model access](/docs/collaborate/govern/model-contracts) for more info.

```

File: website/docs/docs/collaborate/govern/model-contracts.md
Comment (mirnawong1) [inline website/docs/docs/collaborate/govern/model-contracts.md]:
```suggestion
When you're deciding between using a test or a constraint, using a constraint can save on overall costs and computing. However, there are some prerequisites you need to meet before replacing a test with a constraint:
```

File: website/docs/docs/collaborate/govern/model-contracts.md
Comment (mirnawong1) [inline website/docs/docs/collaborate/govern/model-contracts.md]:
```suggestion
- Making sure that your data platform can support and enforce the constraint that you need. Most platforms only enforce `not_null`.
```

File: website/docs/docs/collaborate/govern/model-contracts.md
Comment (mirnawong1) [inline website/docs/docs/collaborate/govern/model-contracts.md]:
```suggestion
- Materializing your data as a table or incremental (not a view or ephemeral)
```

File: website/docs/docs/collaborate/govern/model-contracts.md
Comment (mirnawong1) [inline website/docs/docs/collaborate/govern/model-contracts.md]:
```suggestion
- Defining a contract for this model by specifying the `name` and `data_type` of each column.
```

File: website/docs/reference/commands/parse.md
Comment (mirnawong1) [inline website/docs/reference/commands/parse.md]: we're reducing (or eliminating) our use of 'overview' as the intro header as the user will see the title of the page, followed by  the h2.
Screenshot 2023-04-24 at 16 01 33

```suggestion
```

File: website/docs/guides/migration/versions/02-upgrading-to-v1.5.md
Comment (mirnawong1) [inline website/docs/guides/migration/versions/02-upgrading-to-v1.5.md]:
```suggestion
- More granular [configurations](/reference/global-configs) for logging (to set log format, log levels, and colorization) and cache population
```

File: website/docs/reference/commands/parse.md
Comment (mirnawong1) [inline website/docs/reference/commands/parse.md]:
```suggestion
It will also produce an artifact with detailed timing information, which is useful to understand parsing times for large projects. Refer to [Project parsing](/reference/parsing) for more info.
```

File: website/docs/reference/commands/parse.md
Comment (mirnawong1) [inline website/docs/reference/commands/parse.md]:
```suggestion
Starting in v1.5, `dbt parse` will write or return a [manifest](/reference/artifacts/manifest-json), enabling you to introspect dbt's understanding of all the resources in your project.
```

File: website/docs/reference/resource-configs/contract.md
Comment (mirnawong1) [inline website/docs/reference/resource-configs/contract.md]:
```suggestion
- Additional [`constraints`](resource-properties/constraints), as supported for this materialization and data platform
```

File: website/docs/reference/resource-configs/contract.md
Comment (mirnawong1) [inline website/docs/reference/resource-configs/contract.md]:
```suggestion
This is to ensure that downstream queries of your model (both within and outside dbt) have a predictable and consistent set of columns to use in their analyses. Even a subtle change in data type, such as from `boolean` (`true`/`false`) to `integer` (`0`/`1`), could cause queries to fail in surprising ways.
```

File: website/docs/reference/resource-configs/contract.md
Comment (mirnawong1) [inline website/docs/reference/resource-configs/contract.md]:
```suggestion
That said, when comparing data types in dbt, it doesn't take into account certain details such as size, precision, or scale in its data type comparison. The difference between `varchar(256)` and `varchar(257)` isn't significant enough to affect the experience of downstream queriers. If you need a more precise comparison, it's possible to accomplish it with a custom test.

```

**Comment by mirnawong1** [comment]
hey @jtcohen6 - i've left some suggestions in terms of clarifying or simplifying language but i also see @dbeatty10 has approved it! i don't think my comments are blockers and I've verified the links in the deploy preview.

File: website/docs/reference/global-configs.md
Comment (mirnawong1) [reply website/docs/reference/global-configs.md]: i think our style guide examples dont have the `$` symbol so i agree with @dbeatty10 ! will open an issue!

**Comment by dbeatty10** [comment]
@mirnawong1 I view any suggestions you make as superseding my approval, so please always know I'll never mind if you add blocking comments after I've approved, especially since our eyes are peeled for different types of things! 👀

File: website/docs/guides/migration/versions/02-upgrading-to-v1.5.md
Comment (jtcohen6) [reply website/docs/guides/migration/versions/02-upgrading-to-v1.5.md]: good point - I'll try to strike this balance:
- it's not a prerequisite for upgrading
- we strongly recommend you do it, so that you can take advantage of this feature - whether now or at some point in the future :)

File: website/docs/reference/global-configs.md
Comment (jtcohen6) [reply website/docs/reference/global-configs.md]: everything is ruined by `$`

Comment (jtcohen6) [reply website/docs/reference/global-configs.md]: I'm just going to change them all! try and stop me!

File: website/docs/reference/resource-configs/contract.md
Comment (jtcohen6) [reply website/docs/reference/resource-configs/contract.md]: > If you need a more-precise comparison, it's possible to accomplish with a custom test.

I was being a touch lazy here, but I think the point stands! I'll link to https://docs.getdbt.com/guides/best-practices/writing-custom-generic-tests

File: website/docs/docs/collaborate/govern/model-contracts.md
Comment (jtcohen6) [reply website/docs/docs/collaborate/govern/model-contracts.md]: good call!

File: website/docs/reference/resource-configs/contract.md
Comment (jtcohen6) [reply website/docs/reference/resource-configs/contract.md]: I'm going to switch `downstream queriers` &rarr; `people querying your model downstream`. I do think it's important that we're talking about people here, and not just impersonal queries/systems.

> both within and outside dbt

Agree with "both." I'm going to keep this parenthetical inside m-dashes for emphasis, rather than just an aside, because I think it's a really important point. Doug seems to agree :) https://github.com/dbt-labs/docs.getdbt.com/pull/3249#discussion_r1175370722

File: website/docs/reference/resource-configs/contract.md
Comment (jtcohen6) [reply website/docs/reference/resource-configs/contract.md]: Could I have special dispensation to keep this one a bit more personal/informal in tone? This is our opinion as practitioners (&rarr; dbt maintainers), not an absolute truth

Personal/informal:
> We don't think you should sweat the difference between `varchar(256)` and `varchar(257)`, because it almost never affects the experience of downstream queriers.

Impersonal/formal:
> The difference between `varchar(256)` and `varchar(257)` is really not significant enough to affect the experience of downstream queriers.

Comment (mirnawong1) [reply website/docs/reference/resource-configs/contract.md]: for sure no worries and i love informal tones!! i was only thinking of the word 'sweat' with regard to future translation to be honest so that's why i tried prepping it.

Comment (jtcohen6) [reply website/docs/reference/resource-configs/contract.md]: totally fair! it'll be a fun bridge to cross when we get there

> on pense que la difference entre `varchar(256)` et `varchar(257)` n'est pas très grave, t'inquiète minot

Comment (mirnawong1) [reply website/docs/reference/resource-configs/contract.md]: haha oui oui, c'est pas grave!

**Comment by jtcohen6** [comment]
@dbeatty10 @mirnawong1 Thank you so much for the thorough reviews !! I believe I've responded to & resolved all comments

File: website/docs/docs/collaborate/govern/model-access.md
Comment (mirnawong1) [inline website/docs/docs/collaborate/govern/model-access.md]:
```suggestion
**Model groups and access** and **user groups and access** mean two different things. "User groups and access" is a specific term used in dbt Cloud to manage permissions. Refer to [User access](/docs/cloud/manage-access/about-user-access) for more info.
```

Comment (mirnawong1) [reply website/docs/docs/collaborate/govern/model-access.md]: hey @jtcohen6 - i updated and commited this url as it had a github one - hope you're ok with that!

Comment (jtcohen6) [reply website/docs/docs/collaborate/govern/model-access.md]: whoops!! yes, totally ok, thank you for fixing 🙏

**Comment by mirnawong1** [comment]
this is an awesome piece @jtcohen6 - thank you for letting me take part with @dbeatty10 and i'm excited for users to read through this 🙏 🎉 !

**Review by mirnawong1 — APPROVED** [review]
🚢

**Pull request merged**
